### PR TITLE
chore: add null checks to IAM snippets

### LINKED ISSUE #8176: iam.snippets.QuickstartTests: testQuickstart failed
Note: #6483 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: 746af47f7b8b848c02936459578c5daf280efa1c
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/089ccc40-e331-4df2-9d70-045f4dd3ec55), [Sponge](http://sponge2/089ccc40-e331-4df2-9d70-045f4dd3ec55)
status: failed
<details><summary>Test output</summary><br><pre>java.lang.NullPointerException
	at iam.snippets.Quickstart.setPolicy(Quickstart.java:170)
	at iam.snippets.Quickstart.addBinding(Quickstart.java:122)
	at iam.snippets.QuickstartTests.testQuickstart(QuickstartTests.java:126)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
</pre></details>

### PR DESCRIPTION
**Pull request by kweinmeister**
Fixes #8176

### CODE REVIEW

File: iam/api-client/src/test/java/iam/snippets/QuickstartTests.java [hunk 1]
```diff
@@ -108,7 +108,9 @@ public void tearDown() {
 
     String resource = "projects/-/serviceAccounts/" + serviceAccount.getEmail();
     try {
-      iamService.projects().serviceAccounts().delete(resource).execute();
+      if (resource != null) {
```
Comment (rsamborski) [inline iam/api-client/src/test/java/iam/snippets/QuickstartTests.java:111]: I don't think `resource` can be ever `null` as we declare and assign this variable in line 109.

**Review by rsamborski — COMMENTED** [review]
One nit from my side. @Sita04 PTAL before we merge this.

File: iam/api-client/src/main/java/iam/snippets/Quickstart.java [hunk 2]
```diff
@@ -135,7 +135,7 @@ public static void removeMember(
         break;
       }
     }
-    if (binding.getMembers().contains(member)) {
+    if (binding != null && binding.getMembers().contains(member)) {
```
Comment (Sita04) [inline iam/api-client/src/main/java/iam/snippets/Quickstart.java:138]: Could you also introduce a null check in line 103?

**Review by Sita04 — APPROVED** [review]
Thanks Karl. LGTM.
One minor comment.

**Pull request merged**